### PR TITLE
Set model to not receive documents when empty is selected

### LINKED ISSUE #948: 🐛 Bug Report: Empty document uses default document
### 📜 Description

When Empty document is selected documents from pre-uploaded DocsGPT documentation (Default option) is used. 

### 👟 Reproduction steps

1. Choose Empty from Source Doc selector.
2. Ask any question, you will see that DocsGPT still uses Default source doc. 

### 👍 Expected behavior

Upon selecting Empty, model should not receive any documents. 
The behaviour should be the same as if Chunks processed per query is set to 0. 

### 👎 Actual Behavior with Screenshots

Model receives Default documents. 

### 💻 Operating system

Linux

### What browsers are you seeing the problem on?

_No response_

### 🤖 What development environment are you experiencing this bug on?

Docker

### 🔒 Did you set the correct environment variables in the right path? List the environment variable names (not values please!)

_No response_

### 📃 Provide any additional context for the Bug.

_No response_

### 📖 Relevant log output

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🔗 Are you willing to submit PR?

None

### 🧑‍⚖️ Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by areebniyas**
**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

- Set model to not receive documents when empty is selected

**Why was this change needed?** (You can also link to an open issue here)

- Closes #948

### CODE REVIEW

**Comment by ManishMadan2882** [comment]
Hey @areebniyas 
Think the issue is not in the frontend, as the same payload is sent to the POST /stream.
Can make sure that it needs to be fixed in the backend(/application), as the class selected for none doc type is `ClassicRAG` and the chunks are also 2 which is also same for default selection.
In the issue description it is mentioned that the
`The behaviour should be the same as if Chunks processed per query is set to 0.`
Thanks!

**Comment by areebniyas** [comment]
Hi @ManishMadan2882 , thank you for the input. Would any of below approaches be correct:
1. setting the default to `None`
2. or removing the default retriever

at: https://github.com/arc53/DocsGPT/blob/4dd0d65db432741adea8c52176367d1af6e14a75/application/retriever/retriever_creator.py#L12

**Comment by ManishMadan2882** [comment]
Actually I feel like we might need a fix in the /stream route
There is certain ambiguity in this route, we have active_docs and selected_docs, the condition which we are checking in this [line](https://github.com/arc53/DocsGPT/blob/88623754cfeba9a2d65d6fad44d0c2a13126d330/application/api/answer/routes.py#L306) should actually be checked for active_docs. All we need to do is:
* remove the selected_docs and stick to active_docs only
* pass null/None in case the None document is selected, so that we can remove isNoneDoc(from the frontend)
* The behaviour of None selection should be such that a ClassicRAG works with 0 chunks
* Considering the approaches you mentioned, we don't need to remove any of the retrievers or defaulting to null.
Thanks and please feel free to ask anything regarding this issue.

**Comment by areebniyas** [comment]
Hi @ManishMadan2882 , added some changes you mentioned - [cb71729](https://github.com/arc53/DocsGPT/pull/1347/commits/cb71729545c0242d58ee0cd03c3ca7ed7910c05f). Let me know your feedback and what I could do next.
Thanks for the support.

**Review by ManishMadan2882 — CHANGES_REQUESTED** [review]
This doesn't seem to fix the issue, I tried few things and feel like a simpler way to handle is to add one condition check just before creating the retriever instance.
Something like this:
```
if "isNoneDoc" in data and data["isNoneDoc"] is True:
                chunks = 0
```
For now keep the `isNoneDoc`, but you can remove "selectedDocs" from the stream post(), as we don't use it anywhere further.
Thanks!

**Comment by areebniyas** [comment]
@ManishMadan2882 Added requested changes via [724660e](https://github.com/arc53/DocsGPT/pull/1347/commits/724660e5774a90456ccc1283e35c2af8963ae22d). Anything I missed?

File: application/api/answer/routes.py [hunk 1]
```diff
@@ -303,11 +300,6 @@ def post(self):
             history = json.loads(history)
             conversation_id = data.get("conversation_id")
             prompt_id = data.get("prompt_id", "default")
-            if "active_docs" in data and data["active_docs"] is None:
-                chunks = 0
-                source = {}
-            else:
-                chunks = int(data.get("chunks", 2))
```
[the remark this answers is not in the document]
Comment (areebniyas) [reply application/api/answer/routes.py:309]: Resolved with [3d650d0](https://github.com/arc53/DocsGPT/pull/1347/commits/3d650d0fbffabe402954ebfc71e56ebb854ddcd6)

**Comment by ManishMadan2882** [comment]
@areebniyas there is an error, did you test it on your system before pushing?

**Pull request closed without merging**